rules/scala/zinc/ZincRunner.scala

```diff
@@ -22,12 +22,34 @@ import scala.util.control.NonFatal
 import xsbti.compile.{FileAnalysisStore => _, _}
 import xsbti.{Logger, Reporter}
 
+/**
+ * <strong>Caching</strong>
+ *
+ * Zinc has two caches:
+ *  1. a ClassLoaderCache which is a soft reference cache for classloaders of Scala compilers.
+ *  2. a CompilerCache which is a hard reference cache for (I think) Scala compiler instances.
+ *
+ * The CompilerCache has reproducibility issues, so it needs to be a no-op.
+ * The ClassLoaderCache needs to be reused else JIT reuse (i.e. the point of the worker strategy) doesn't happen.
+ *
+ * There are two sensible strategies for Bazel workers
+ *  A. Each worker compiles multiple Scala versions. Trust the ClassLoaderCache's timestamp check. Maintain a hard
```

Comment:
The compiler cache cache is never populated by default, it was added to support resident compilation, which doesn't work nowadays.